### ISSUE TITLE
Visibility changes for PhpOffice\PhpSpreadsheet\Helper\Html breaks subclasses

### ISSUE THREAD

**Issue by jwoodhead** [PHPOffice/PhpSpreadsheet#3751]
This is:

```
- [X] a bug report
- [ ] a feature request
- [ ] **not** a usage question (ask them on https://stackoverflow.com/questions/tagged/phpspreadsheet or https://gitter.im/PHPOffice/PhpSpreadsheet)
```

### What is the expected behavior?
The properties/methods on the `PhpOffice\PhpSpreadsheet\Helper\Html` class were previously set as protected which allowed for implementing custom functionality via a child classes.

### What is the current behavior?
The visibility for the properties/methods was changed to private as part of #3080, breaking any child classes that were previously extending it.

### What are the steps to reproduce?

```php
<?php

require __DIR__ . '/vendor/autoload.php';

// Create new Spreadsheet object
$spreadsheet = new \PhpOffice\PhpSpreadsheet\Spreadsheet();

// add code that show the issue here...

class ChildHtmlHelper extends \PhpOffice\PhpSpreadsheet\Helper\Html
{
    // this is referring to properties and methods that are no longer valid
    protected function initialise(): void
    {
        parent::initialise();

        $this->startTagCallbacks = array_merge(
            $this->startTagCallbacks,
            [
                'ul' => 'breakTag',
                'li' => 'startLiTag',
            ]
        );
        $this->endTagCallbacks = array_merge(
            $this->endTagCallbacks,
            [
                'ul' => 'breakTag',
                'li' => 'breakTag',
            ]
        );
    }

    protected function startLiTag($tag): void
    {
        $this->stringData .= "\u{00A0}\u{2022} \u{00A0}";
    }
}

$exampleHtml = '<ul><li>List Item 1</li><li>List Item 2</li></ul>';

$richTextObject = (new ChildHtmlHelper)->toRichTextObject($exampleHtml);

$spreadsheet->getActiveSheet()->setCellValue('A1', $richTextObject);

$writer = new \PhpOffice\PhpSpreadsheet\Writer\Xlsx($spreadsheet);

$writer->save('example_excel.xlsx');
```

### What features do you think are causing the issue

- [ ] Reader
- [X] Writer
- [ ] Styles
- [ ] Data Validations
- [ ] Formula Calculations
- [ ] Charts
- [ ] AutoFilter
- [ ] Form Elements

### Does an issue affect all spreadsheet file formats? If not, which formats are affected?
- XLSX
- HTML
- PDF

### Which versions of PhpSpreadsheet and PHP are affected?
PhpSpreadsheet versions >1.26.0

**Comment by oleibman** [comment]
This is an issue that has arisen, and has not been acted on, before, e.g. see #3348 and especially #1682. Missing from those items is a use case to demonstrate why the visibility change would be helpful. You have provided one (using a different bulleting style than the default for lists), and it seems like a reasonable request.

We could restore the visibility in order for you to achieve your aims. However, we could also just, say, add a property `liBullet`, add `li` to the arrays, and add a method which handles list items based on the value in liBullet. That would be simpler and more reliable for us, and probably for you. If there are few cases where you think the current implementation has gaps, that would probably be a better way to go. However, if you think there's lots of gaps in the current implementation, perhaps allowing subclassing is better. If you think the latter is the case, it would be helpful to let me know what you think some of the other missing use cases might be.

**Comment by jwoodhead** [comment]
I do agree that the visibility for the methods/properties should not be restored.

There is no handling for lists in the current implementation which is why we implemented our own solution.

Another gap we discovered is handling for tables. We use `PhpSpreadsheet` with user-generated WYSIWYG content and implemented our own solution to convert tables into a format that is "legible" using non-breaking spaces and line breaks so each cell is on it's own line. We consider this to be a bodge but it works for our specific implementation.

Adding support for tables to the library would be nice, but I imagine there isn't going to be a "best" solution for how to render a table as text in a cell.

One option that might work for allowing customization is implementing static methods for adding start or end tag callbacks as closures. 

```php
\PhpOffice\PhpSpreadsheet\Helper\Html::addStartTagCallback('table', function($stringData) {
    return $stringData."\n";
}
```
